### PR TITLE
fix containerd config generation with go template

### PR DESCRIPTION
**Pull request by vflaux**
This is an alternative proposal to fix https://github.com/spegel-org/spegel/issues/549 using go template instead of a complicated struct generation (https://github.com/spegel-org/spegel/pull/550).
The tests are the same as in https://github.com/spegel-org/spegel/pull/550.

### CODE REVIEW

File: pkg/oci/containerd.go [hunk 1]
```diff
@@ -475,3 +504,33 @@ func getHostFile(fs afero.Fs, configPath string, appendToBackup bool, registryUR
 	}
 	return hf, false, nil
 }
+
+// getOrderedHosts parses the given byte slice as TOML data and returns a slice of
+// strings containing the ordered hosts found in the TOML data.
+func getOrderedHosts(b []byte) []string {
```
Comment (phillebaba) [inline pkg/oci/containerd.go:510]: Do we still need this? Could this be asserted through string comparison in tests?

**Review by phillebaba — CHANGES_REQUESTED** [review]
Most of this looks good. I like the tests and it solves the problem in a simple way. As long as we remove any use of the toml library from the code I will be fine with merging tthis,

File: pkg/oci/containerd_test.go [hunk 2]
```diff
@@ -466,3 +481,48 @@ func stringListToUrlList(t *testing.T, list []string) []url.URL {
...
+[host.'http://127.0.0.5:5000']
+capabilities = ['pull', 'resolve']
+
+[host.'http://127.0.0.1:5000']
+capabilities = ['pull', 'resolve']
+`)},
+			want: []string{"http://127.0.0.4:5000", "http://127.0.0.3:5000", "http://127.0.0.2:5000", "http://127.0.0.5:5000", "http://127.0.0.1:5000"},
+		},
+	}
+	for _, tt := range tests {
```
Comment (phillebaba) [inline pkg/oci/containerd_test.go:522]: Add `t.Parallel()`

File: pkg/oci/containerd_test.go [hunk 3]
```diff
@@ -466,3 +481,48 @@ func stringListToUrlList(t *testing.T, list []string) []url.URL {
 	}
 	return urls
 }
+
+func TestGetOrderedHosts(t *testing.T) {
```
Comment (phillebaba) [inline pkg/oci/containerd_test.go:485]: Add `t.Parallel()`

File: pkg/oci/containerd.go (hunk 1 above)
Comment (vflaux) [reply pkg/oci/containerd.go:510]: Sorry, I don't really understand what you mean.
You need to get the hosts in order to be able to rebuild the config file with the same order of hosts as previously (with spegel hosts added).

**Comment by vflaux** [comment]
> As long as we remove any use of the toml library from the code I will be fine with merging tthis,

The toml lib would still be used to unmarshall the config even if I implement a `getOrderedHosts()` without it and don't use it in the go template.

**Comment by phillebaba** [comment]
Closing as this has been fixed in #601

**Pull request closed without merging**